src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -0,0 +1,323 @@
+                    {},
+                );
+            },
+            (error: HttpErrorResponse) => {
+                if (error.status === 404) {
+                    // there are no unassessed submission, nothing we have to worry about
+                    this.jhiAlertService.error('artemisApp.tutorExerciseDashboard.noSubmissions');
+                } else {
+                    this.onError(error.message);
+                }
+            },
+        );
+    }
+
+    /**
+     * Sends the current (updated) assessment to the server to update the original assessment after a complaint was accepted.
+     * The corresponding complaint response is sent along with the updated assessment to prevent additional requests.
+     *
+     * @param complaintResponse the response to the complaint that is sent to the server along with the assessment update
+     */
+    onUpdateAssessmentAfterComplaint(complaintResponse: ComplaintResponse): void {
+        this.manualResultService.updateAfterComplaint(this.manualResult!.feedbacks, complaintResponse, this.manualResult!, this.manualResult!.submission!.id).subscribe(
+            (result: Result) => {
+                this.manualResult = result;
+                this.jhiAlertService.clear();
+                this.jhiAlertService.success('artemisApp.assessment.messages.updateAfterComplaintSuccessful');
+            },
+            () => {
+                this.jhiAlertService.clear();
+                this.jhiAlertService.error('artemisApp.assessment.messages.updateAfterComplaintFailed');
+            },
+        );
+    }
+
+    navigateBack() {
+        assessmentNavigateBack(this.location, this.router, this.exercise, this.submission);
+    }
+
+    /**
+     * Boolean which determines whether the user can override a result.
+     * If no exercise is loaded, for example during loading between exercises, we return false.
+     * Instructors can always override a result.
+     * Tutors can override their own results within the assessment due date, if there is no complaint about their assessment.
+     * They cannot override a result anymore, if there is a complaint. Another tutor must handle the complaint.
+     */
+    get canOverride(): boolean {
+        if (this.exercise) {
+            if (this.isAtLeastInstructor) {
+                // Instructors can override any assessment at any time.
+                return true;
+            }
+            if (this.complaint && this.isAssessor) {
+                // If there is a complaint, the original assessor cannot override the result anymore.
+                return false;
+            }
+            let isBeforeAssessmentDueDate = true;
+            // Add check as the assessmentDueDate must not be set for exercises
+            if (this.exercise.assessmentDueDate) {
+                isBeforeAssessmentDueDate = moment().isBefore(this.exercise.assessmentDueDate!);
```

Comment:
The exclamation mark at the end should not be necessary

Reply:
done